src/Welcome.js

```diff
@@ -0,0 +1,24 @@
+import React, { useEffect } from 'react';
+import getToken from './lib/tokens';
+import { useHistory } from 'react-router-dom';
+
+const Welcome = () => {
+  const history = useHistory();
+  useEffect(() => {
+    if (localStorage.getItem('token')) {
+      history.push('/list');
+    }
+  }, [history]);
+  const handleClick = () => {
+    const token = getToken();
+    localStorage.setItem('token', token);
+    history.push('/list');
+  };
```

Comment:
Nice work here! I think it would be a good decision to take all the code here and lift it up to either `App.js` OR into it's own custom hook. Moving that functionality up to a higher level should allow us to only render all the other routes only if there is a token. What do you think about that? Let me know how that works out or if you have any questions!

Reply:
Seems like you can't use history inside the App component because it's not a Route itself. I did things differently -- created a PrivateRoute that redirects to welcome page if the token is not set.